Tangem/Modules/Swapping/SwappingViewModel.swift

```diff
@@ -172,23 +174,34 @@ private extension SwappingViewModel {
         coordinator.presentSuccessView(inputModel: inputModel)
     }
 
-    @MainActor
     func openPermissionView() {
         let state = exchangeManager.getAvailabilityState()
+        let source = exchangeManager.getExchangeItems().source
+
         guard case .available(let result, let info) = state,
-              result.isPermissionRequired else {
+              result.isPermissionRequired,
+            fiatRatesProvider.hasRates(for: source) else {
+            // If we don't have enough data disable button and refresh()
+            mainButtonIsEnabled = false
+            exchangeManager.refresh()
+
             return
         }
 
-        let inputModel = SwappingPermissionInputModel(
-            fiatFee: result.fiatFee,
-            transactionInfo: info
-        )
+        Task {
+            let fiatFee = try await fiatRatesProvider.getFiat(for: source, amount: info.fee)
```

Comment:
все же лучше использовать эту конструкцию, т.к. реализация из коробки может вызвать утечки... https://tangem.slack.com/archives/C01FQL72BLL/p1673244453177069
```suggestion
        runTask { [weak self] in
            guard let self else { return }
            
            let fiatFee = try await fiatRatesProvider.getFiat(for: source, amount: info.fee)
```

Reply:
это везде так делать будем ? Это ж беда какая то получается )

Reply:
именно на 190 строке сделал через runTask, и добавил метод с захватом что бы проверку не писать